frontend/test/pytest/test_preprocess.py

```diff
@@ -337,32 +337,64 @@ def circuit(theta: float):
         assert "expval" not in mlir
         assert "counts" in mlir
 
-    def test_non_commuting_measurements_are_split(self, mocker):
-        """Test that the split_non_commuting transform is added to the transform
-        program from preprocess when non_commuting_observables_flag is False"""
+    def test_measurements_are_split(self, mocker):
+        """Test that the split_to_single_terms or split_non_commuting transform
+        are added to the transform program from preprocess as expected, based on the
+        sum_observables_flag and the non_commuting_observables_flag"""
 
-        # dummy device supports non_commuting observables by default
         dev = DummyDevice(wires=4, shots=1000)
-
-        # Create a qjit device that supports non-commuting observables
         dev_capabilities = get_device_capabilities(dev, ProgramFeatures(dev.shots is not None))
+
+        # dev1 supports non-commuting observables and sum observables - no splitting
+        assert "Sum" in dev_capabilities.native_obs
+        assert "Hamiltonian" in dev_capabilities.native_obs
+        assert dev_capabilities.non_commuting_observables_flag is True
         backend_info = extract_backend_info(dev, dev_capabilities)
         qjit_dev1 = QJITDeviceNewAPI(dev, dev_capabilities, backend_info)
 
-        # Create a qjit device that does NOT support non-commuting observables
-        dev_capabilities = replace(dev_capabilities, non_commuting_observables_flag=False)
+        # dev2 supports non-commuting observables but NOT sums - split_to_single_terms
+        del dev_capabilities.native_obs["Sum"]
+        del dev_capabilities.native_obs["Hamiltonian"]
         backend_info = extract_backend_info(dev, dev_capabilities)
         qjit_dev2 = QJITDeviceNewAPI(dev, dev_capabilities, backend_info)
 
+        # dev3 supports does not support non-commuting observables OR sums - split_non_commuting
+        dev_capabilities = replace(dev_capabilities, non_commuting_observables_flag=False)
+        backend_info = extract_backend_info(dev, dev_capabilities)
+        qjit_dev3 = QJITDeviceNewAPI(dev, dev_capabilities, backend_info)
+
+        # dev4 supports sums but NOT non-commuting observables - split_non_commuting
+        dev_capabilities = replace(dev_capabilities, non_commuting_observables_flag=False)
```

Comment:
I wonder will this case split the sums even though they could run on the device?

Edit: And would such a configuration even make sense? 😅 I suppose it could be a sum of only commuting terms 🤔

Reply:
Yes, this transform is not clever enough to notice that a particular sum is only commuting terms and leave it alone. It indiscriminately splits up all the sums and distributes their terms (though it would put them all on the same tape if they all commute)